addon_service/common/osf.py

```diff
@@ -58,12 +58,7 @@ async def get_osf_user_uri(request: django_http.HttpRequest) -> str | None:
     _auth_headers = _get_osf_auth_headers(request)
     if not _auth_headers:
         return None
-    _client = await get_singleton_client_session()
-    async with _client.get(_osfapi_me_url(), headers=_auth_headers) as _response:
-        if HTTPStatus(_response.status).is_client_error:
-            return None
-        _response_content = await _response.json()
-        return _iri_from_osfapi_resource(_response_content["data"])
+    return request.session["user_reference_uri"]
```

Comment:
I see. [Here](https://github.com/CenterForOpenScience/osf.io/pull/10965/files#diff-eb9961a790efef49b6d3b1039635e19bd3e21d0664c88df3318ffde221a49afeR38) is the OSF piece that support your change:

> 'user_reference_uri': user.get_semantic_iri(),

One question, should we replace this piece as well?

```python
_auth_headers = _get_osf_auth_headers(request)
if not _auth_headers:
    return None
````

e.g. just check if `request.session` has a valid `user_reference_uri`?

Finally, looking at `_get_osf_auth_headers()`, does this session stuff break requests using personal access token?